client/my-sites/controller.js

```diff
@@ -121,11 +120,11 @@ function renderSelectedSiteIsDomainOnly( reactContext, selectedSite ) {
 			title: i18n.translate( 'This feature is not available for domains' ),
 			line: i18n.translate( 'To use this feature you need to create a site' ),
 			action: i18n.translate( 'Create New Site' ),
-			actionURL: `/start/site-selected/?siteSlug=${ encodeURIComponent( selectedSite.slug ) }`,
+			actionURL: `/start/site-selected/?siteSlug=${ encodeURIComponent( selectedSite.slug ) }&siteId=${ encodeURIComponent( selectedSite.ID ) }`,
```

Comment:
We should always encode parameters, I guess in that specific case it doesn't matter, but if you don't mind I'd like to keep it.